Adds medical resident role

DeadmonWonderland opened the pull request:
Resident as an alt title has good intentions but it still has It's issues. We used to have 3 Physicians at all times, with this change we would have the possibility of two plus one trainee. This change should hopefully make the resident role a better role for people to learn surgical mechanics and such.

:cl: Deadmon
rscadd: Adds Medical Resident title, sort of like a physician lite role. Mostly identical in mechanics to Physician, with the exception of a slightly lower minimum age requirement and less overall points.
rscdel: Removes Resident alt-title from Physician.
/:cl:

Flying-loulou commented:
You need to make/set an HUD icon for your new job, this shall appease Travis.

Flying-loulou commented:
You need to make/set an HUD icon for your new job, this shall appease Travis.

DeadmonWonderland commented:
Pull failed due to
```
## UNIT_TEST ##: [SUBSYSTEM - ATOMS: Shall have no bad init calls]: Path : /obj/structure/lattice 
- Qdel'd in New()
## UNIT_TEST ##: !!! FAILURE !!! [SUBSYSTEM - ATOMS: Shall have no bad init calls]: Atoms had bad initialization calls.
```

Which has nothing to do with my PR and I'm not sure what's going on with it.

DeadmonWonderland commented:
Travis fixed themselves it seems

PsiOmegaDelta commented on maps/torch/job/medical_jobs.dm:
```diff
@@ -42,6 +41,46 @@
 	software_on_spawn = list(/datum/computer_file/program/suit_sensors,
 							 /datum/computer_file/program/camera_monitor)
 
+/datum/job/junior_doctor
```
Seems to me like this could be a subtype of `/datum/job/doctor`, ie. `/datum/job/doctor/junior` and then only override fields which are different

PsiOmegaDelta replied:
Not related to you, but having looked at this code I have to wonder why there isn't a base `/datum/job/medical` type, shared by the roles in this file

DeadmonWonderland replied:
I can rewrite it to be that if you want.

DeadmonWonderland replied:
I can rewrite it to be that if you want.

DeadmonWonderland replied:
I can rewrite it to be that if you want.

DeadmonWonderland commented:
looks like something fucked, one sec

DeadmonWonderland committed:
Adds Medical Resident, Fixed

Adds Medical resident, a lower age required medical role. Has same skillset as physician.
